core/src/main/java/org/apache/accumulo/core/client/security/tokens/KerberosToken.java

```diff
@@ -51,11 +52,11 @@
    *          The user that is logged in
    */
   public KerberosToken(String principal) throws IOException {
-    requireNonNull(principal);
+    this.principal = requireNonNull(principal);
     final UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-    checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
-    checkArgument(principal.equals(ugi.getUserName()), "Provided principal does not match currently logged-in user");
-    this.principal = ugi.getUserName();
+    if (AuthenticationMethod.KERBEROS == ugi.getAuthenticationMethod()) {
+      checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
+    }
```

Comment:
FWIW, the current code would always fail without Kerberos credentials, so it was already somewhat comprehensive, covering the other cases (SIMPLE, TOKEN, etc.), so I wouldn't go so far as to say your previous logic was sub-par (the issue in this PR aside). But, after this change, the extra check would be nice to restore the previous restrictions to the Kerberos and now Proxy case.